### PR TITLE
build(node): Support Node 16

### LINKED ISSUE #1582: [fix] Node Engine Requirement?
**ReactUSWDS Version & USWDS Version:**
ReactUSWDS: v2.0.0
USWDS: n/a

**Describe the bug**
Unable to install when using a node engine >14. Unsure if this is a bug or an undocumented requirement.

**To Reproduce**
`npm i @trussworks/react-uswds` with node@16.5.0 (LTS changes to v16 on 2021-10-26)

out:
```bash
❯ npm i @trussworks/react-uswds
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: @trussworks/react-uswds@2.0.0
npm ERR! notsup Not compatible with your version of node/npm: @trussworks/react-uswds@2.0.0
npm ERR! notsup Required: {"node":"10.x - 14.x"}
npm ERR! notsup Actual:   {"npm":"7.19.1","node":"v16.5.0"}
```

**Expected behavior**
Successful install

_Thanks for putting together this library, I'm excited to try it out 😄_

### PR DESCRIPTION
**Pull request by brandonlenz**
# Summary

Add support for Node 16 (pushed to internal branch from forked PR)

## Related Issues or PRs

Internally branched PR for @epicfaace's change in https://github.com/trussworks/react-uswds/pull/1743/ to allow our CI to run.

> Fixes #1582, so that users with node 16 can install this package.

### CODE REVIEW

**Comment by brandonlenz** [comment]
Question to reviewers:

Should we update the lower end of the range to be 12.x since 10.x [is no longer maintained](https://nodejs.org/en/about/releases/)?

If so, that would make this a potentially breaking change, correct?

**Comment by trussworks-infra** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Changes were made to package.json, but not to yarn.lock - <i>Perhaps you need to run `yarn install`?</i>
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 79e00640b59ddca536804be16b3cc2a8a92cdcca
</p>

**Comment by PDickeyTruss** [comment]
👋 Hi, just wanted to make a little noise on this and see if there was anything needed to get this PR pushed through. I am hoping to use react-uswds with node 16 😉

**Review by suzubara — APPROVED** [review]
I made some updates to make sure we run CI tests on Node 12, 14, and 16. I think we should drop 10 support in a different PR that is, yes, a breaking change.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -30,16 +35,33 @@ jobs:
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12.x
```
Comment (brandonlenz) [inline .github/workflows/build-and-test.yml:48]: @suzubara should we use 12 here, or default to a newer release if we're intending to support up to 16?

Comment (suzubara) [reply .github/workflows/build-and-test.yml:48]: IMO in the interest of changing one thing at a time, we could bump this to 16 in https://github.com/trussworks/react-uswds/pull/1778

Comment (brandonlenz) [reply .github/workflows/build-and-test.yml:48]: Ooof. I made these so long ago, I forgot I had that other PR to make that change 😆 Thanks!

Comment (brandonlenz) [reply .github/workflows/build-and-test.yml:48]: Ooof. I made these so long ago, I forgot I had that other PR to make that change 😆 Thanks!

**Pull request merged**